tsdb/metrics.go

```diff
@@ -0,0 +1,227 @@
+package tsdb
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sort"
+)
+
+// namespace is the leading part of all published metrics for the Storage service.
+const namespace = "storage"
+
+// Metrics wrap all collectors of storage-related metrics.
+type Metrics struct {
+	GlobalLabels *prometheus.Labels
+	TSICache     *TSICacheMetrics
+	TSIPartition *TSIPartitionMetrics
+}
+
+func NewMetrics(globalLabels *prometheus.Labels) *Metrics {
```

Comment:
Looking through the prometheus docs, I'm thinking this `globalLabels` approach might be an anti-pattern... continuing to investigate, but wanted to flag it.

If we ditch `globalLabels` here then we _could_ have all these structs as module-level singletons instantiated by `init()`. That could simplify all the code used to thread references through layers of the storage engine.